R/class-workbook.R

```diff
@@ -4126,11 +4142,88 @@ wbWorkbook <- R6::R6Class(
       value
     },
 
+    validate_new_sheet = function(sheet) {
+      # returns nothing, throws error if there's a problem.
+
+      if (is.na(sheet)) {
+        stop("sheet cannot be NA")
+      }
+
+      if (is.numeric(sheet)) {
+        if (sheet %% 1 != 0) {
+          stop("If sheet is numeric it must be an integer")
+        }
+
+        if (sheet <= 0) {
+          stop("if sheet is an integer it must be a positive number")
+        }
+
+        if (sheet <= length(self$sheet_names)) {
+          stop("there is already a sheet at index ", sheet)
+        }
+
+        return()
```

Comment:
Forgotten?

Reply:
Nope.  This function is just to serves as a validation check, not suppose to be called for assignment

Reply:
ah got it